File: app/src/main/java/com/univreview/model/Subject.java
package com.univreview.model;

import com.google.gson.annotations.Expose;

import java.io.Serializable;

/**
 * Created by DavidHa on 2017. 1. 13..
 */
public class Subject extends AbstractDataProvider  implements Serializable{
    @Expose
    public Long id;
    @Expose
    public String name;
    @Expose
    public String year; //학년
    @Expose
    public int point;   //학점

    @Expose
    public String subjectName;

    @Override
    public long getId() {
        return id;
    }

    @Override
    public String getName() {
        return name;
    }
}

File: app/src/main/java/com/univreview/model/Subject.kt
package com.univreview.model

import android.os.Parcel
import android.os.Parcelable
import com.dhha22.bindadapter.Item

/**
 * Created by DavidHa on 2017. 8. 20..
 */
data class Subject(var id: Long, var name: String) : Parcelable, Item {
    constructor(source: Parcel) : this(
            source.readLong(),
            source.readString()
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) = with(dest) {
        writeLong(id)
        writeString(name)
    }

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<Subject> = object : Parcelable.Creator<Subject> {
            override fun createFromParcel(source: Parcel): Subject = Subject(source)
            override fun newArray(size: Int): Array<Subject?> = arrayOfNulls(size)
        }
    }
}